### ISSUE TITLE
Signal quality/CC errors not populated "randomly" in a minisatip instance connected as client to another minisatip server

### ISSUE THREAD

**Issue by brainstormi** [catalinii/minisatip#788]
Hi...
 Due to reasons about reliability of tvheadend decrypting and certain satip client devices, I had to configure another minisatip instance in the middle acting as "proxy" in some way. Tvh connect to this instance trough TCP and this last one to another minisatip server in a different network (TCP or UDP mode has the same outcome):

TvH ---->  minisatip (proxy/decrypting) ----> minisatip server (RAW TS stream, no decryption) ----> dvb-s2 usb tunner
                                                                  
   First of all the solution run stable, no complains about there. There are some minor things I notice and I expect you can support me to understand it:

- Signal quality when tunning a new channels appears empty in minisatip "proxy" instance most of the times you tune a new channel. In case it works, it continue working until you change the channel, then you "throw the dice" again. 
- CC errors seems not being "forwarded" in all cases from minisatip "server" to minisatip "proxy". Sometimes I see CC errors on server instance, which is sending this stream to the proxy one to be decoded, but in the last one they are not reported, although Tvh seems to detect them.

    If there are any logs you would need it, let me know it which ones to enable.

Regards,

**Comment by catalinii** [comment]
1) Can you check after 3 seconds after the channel is changed? Generally it should take few seconds to see the new signal.
2) CC calculation is done individually by each minisatip/tvh. If you see CC on the tvh but nothing on minisatip maybe that link is the problem.

Generally packets can be lost because of 3 main reasons:
- Weak signal or HW issues
- Using UDP - if you have the option use always TCP 
- For TCP the network buffers are full so data needs to be discarded (this is shows in the UI)

First can appear just on the minisatip that has access to the hardware, the other 2 can appear anywhere.
Related to tvh, make sure tvh uses also TCP

**Comment by brainstormi** [comment]
Hi @catalinii
    Thanks for your quick feedback. Inline with your points:
    1. Signal reporting is never displayed if it doesn't work at the beginning when changing the channel.
    2. Original CC errors reported by minisatip connected to the DVBS2 usb adapter are due to weak signal, nothing can be done there, I agree. The issue is the minisatip instance sitting in the middle between this one and tvheadend, being the only one not detecting in some cases these CC errors, it doesn't matter if it's connected through UDP or TCP, although I discarded network issues, after checking no TCP/UDP errors are reporting, no packet drops or errors after increasing OS buffers values to about 2MB (rmem_max and wmem_max).
    So, basically minisatip "source" detect a CC errors due to Weak Signal, minisatip bridge/proxy sometimes doesn't detect them and tvheadend at the end of this chain, also sees the same CC errors reported by the first feeding all this stream.

TvH ----> minisatip (proxy/decrypting) ----> minisatip server (RAW TS stream, no decryption) ----> dvb-s2 usb tunner

**Comment by lars18th** [comment]
Hi @brainstormi ,

I feel this is resolved with recent versions. Please could you check it for closing this issue?
Thank you.

**Comment by Jalle19** [comment]
Closing this, cannot seem to replicate it anymore
